accounts/service.go

```diff
@@ -41,22 +43,42 @@ func NewService(
 	return &Service{store, km, fc, wp, txs, tes, cfg}
 }
 
-func (s *Service) InitAdminAccount() {
+func (s *Service) InitAdminAccount(ctx context.Context, txService *transactions.Service) error {
 	a, err := s.store.Account(s.cfg.AdminAccountAddress)
 	if err != nil {
 		if !strings.Contains(err.Error(), "record not found") {
-			panic(err)
+			return err
 		}
 		// Admin account not in database
 		a = Account{Address: s.cfg.AdminAccountAddress}
 		err := s.store.InsertAccount(&a)
 		if err != nil {
-			panic(err)
+			return err
 		}
 		AccountAdded.Trigger(AccountAddedPayload{
 			Address: flow.HexToAddress(s.cfg.AdminAccountAddress),
 		})
 	}
+
+	keyCount, err := s.km.InitAdminProposerKeys(ctx)
+	if err != nil {
+		return err
+	}
+
+	if keyCount < s.cfg.AdminProposerKeyCount {
+		numProposers := s.cfg.AdminProposerKeyCount - keyCount
+
+		_, _, err = txService.Create(ctx, true, s.cfg.AdminAccountAddress, templates.Raw{
+			Code: template_strings.AddProposerKeyTransaction,
+			Arguments: []templates.Argument{
+				cadence.NewUInt16(numProposers),
+			},
+		}, transactions.General)
+
+		_, err = s.km.InitAdminProposerKeys(ctx)
+	}
+
+	return err
```

Comment:
```suggestion
		_, err = s.km.InitAdminProposerKeys(ctx)
		if err != nil {
		  return err
		}
	}

	return nil
```

We adopted this in the flow-go codebase a while back, but we prefer to be explicit about error paths and always return from a function as soon as an error occurs. This is mainly to avoid bugs and confusion regarding where errors originate from. It also makes it easier to wrap errors with additional information.